api/server/router/local/image.go

```diff
@@ -284,6 +294,18 @@ func (s *router) postBuild(ctx context.Context, w http.ResponseWriter, r *http.R
 	w.Header().Set("Content-Type", "application/json")
 
 	version := httputils.VersionFromContext(ctx)
+	output := ioutils.NewWriteFlusher(w)
+	sf := streamformatter.NewJSONStreamFormatter()
+	errf := func(err error) error {
+		// Do not write the error in the http output if it's still empty.
+		// This prevents from writing a 200(OK) when there is an interal error.
+		if !output.Flushed() {
+			return err
+		}
+		w.Write(sf.FormatError(errors.New(utils.GetErrorMessage(err))))
+		return nil
```

Comment:
i just replicated the behavior from the existing code. See https://github.com/docker/docker/pull/16147/files#diff-3e7b06b11f8860a92b1bcb4a17ac332dL351